codeframe/cli.py

```
@@ -163,6 +169,94 @@ def agents(
+            try:
+                webbrowser.open(f"http://localhost:{port}")
+            except Exception as e:
+                console.print(f"[yellow]Warning:[/yellow] Could not open browser: {e}")
+                console.print(f"Please open http://localhost:{port} manually")
+
+        browser_thread = threading.Thread(target=open_in_browser, daemon=True)
+        browser_thread.start()
+
+    # Start server (blocking call)
+    try:
+        subprocess.run(cmd, check=True, capture_output=True, text=True)
```

Ruff (0.14.5)

`subprocess` call: check for execution of untrusted input

(S603)